src/kOS/Module/kOSProcessor.cs

```diff
@@ -498,18 +502,19 @@ private void InitProcessorTracking()
                     return (a.part.uid() < b.part.uid()) ? -1 : (a.part.uid() > b.part.uid()) ? 1 : 0;
                 });
             }
-            GameEvents.onPartDestroyed.Add(OnDestroyingMyHardware);
         }
 
-        private void OnDestroyingMyHardware(Part p)
+        public void OnDestroy()
```

Comment:
Yup, very reliable.  That's why I added the super-verbose log the followed, so that I could confirm that it was happening when it was supposed to.